### PR TITLE
Add metrics emission surrounding boltdb performance

### PR DESCRIPTION
**Pull request by mkeeler**
This includes both paths using the legacy boltdb and the newer etcd-io/bbolt in the v2 directory.

The stats available through the Stats method on the DB are emitted as metrics using armon/go-metrics. In addition to those DB stats the StoreLogs method also will measure the time spent storing logs, log size, log batch size and the number of logs per batch.

### CODE REVIEW

File: v2/metrics.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+	metrics "github.com/armon/go-metrics"
+	"go.etcd.io/bbolt"
+)
+
+const (
+	defaultMetricsInterval = 5 * time.Second
+)
+
+func (b *BoltStore) RunMetrics(ctx context.Context, interval time.Duration) {
```
Comment (dnephin) [inline v2/metrics.go:15]: A godoc string on this to mention "this is a long running function expected to be run in a goroutine, will exit when context is cancelled" might be good here.

File: v2/metrics.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	newStats := b.conn.Stats()
+
+	stats := newStats
+	if prev != nil {
+		stats = newStats.Sub(prev)
+	}
+
+	// freelist metrics
+	metrics.SetGauge([]string{"raft", "boltdb", "numFreePages"}, float32(newStats.FreePageN))
+	metrics.SetGauge([]string{"raft", "boltdb", "numPendingPages"}, float32(newStats.FreePageN))
```
Comment (dnephin) [inline v2/metrics.go:44]: Is there a typo here? This appears to be emitted the same value as two names. Should one of these be `stats.FreePageN` ?

**Review by dnephin — COMMENTED** [review]
Nice, I think this looks good.

I had one question and one suggestion. I made them both on the v2 version, but I think they may also apply to the v1.

File: v2/metrics.go (hunk 1 above)
Comment (mkeeler) [reply v2/metrics.go:17]: Good call. Its there now.

File: v2/metrics.go (hunk 2 above)
Comment (mkeeler) [reply v2/metrics.go]: Yes there is. Its fixed now.

**Review by dnephin — APPROVED** [review]
LGTM!

**Pull request merged**